Mathlib/Topology/Category/TopCat/Sphere.lean

```diff
@@ -50,4 +59,32 @@ def diskBoundaryInclusion (n : ℕ) : ∂𝔻 n ⟶ 𝔻 n :=
         rw [isOpen_induced_iff, ← hst, ← hrs]
         tauto⟩ }
 
+/-- The inclusion `𝔹 n ⟶ 𝔻 n` of the interior of the `n`-disk. -/
+def ballInclusion (n : ℕ) : 𝔹 n ⟶ 𝔻 n :=
+  ofHom
+    { toFun := fun ⟨p, hp⟩ ↦ ⟨p, Metric.ball_subset_closedBall hp⟩
+      continuous_toFun := ⟨fun t ⟨s, ⟨r, hro, hrs⟩, hst⟩ ↦ by
+        rw [isOpen_induced_iff, ← hst, ← hrs]
+        tauto⟩ }
+
+instance {n : ℕ} : Mono (diskBoundaryInclusion n) := mono_iff_injective _ |>.mpr <| by
+  intro ⟨x, hx⟩ ⟨y, hy⟩ h
+  simp [diskBoundaryInclusion, disk] at h
+  cases h; rfl
+
+instance {n : ℕ} : Mono (ballInclusion n) := TopCat.mono_iff_injective _ |>.mpr <| by
+  intro ⟨x, hx⟩ ⟨y, hy⟩ h
+  simp [ballInclusion, disk] at h
+  cases h; rfl
+
+instance compact_disk (n : ℕ) : CompactSpace (𝔻 n) := by
+  convert Homeomorph.compactSpace Homeomorph.ulift.symm
+  rw [← isCompact_iff_compactSpace]
+  fapply ProperSpace.isCompact_closedBall
```

Comment:
If we add the missing instance:
```lean
instance {α : Type*} [PseudoMetricSpace α] [ProperSpace α] (x : α) (r : ℝ) :
    CompactSpace (closedBall x r) :=
  isCompact_iff_compactSpace.mp (isCompact_closedBall _ _)
```
just above `Metric.sphere.compactSpace` in the file `Topology.MetricSpace.ProperSpace` then this can become just:
```suggestion
  infer_instance
```